bids/grabbids/bids_layout.py

```diff
@@ -63,10 +49,10 @@ def get_metadata(self, path):
     def get_fieldmap(self, path):
         sub = os.path.split(path)[1].split("_")[0].split("sub-")[1]
         fieldmap_set = {}
-        for file in self.get(subject=sub,
-                             type='(phase1|phase2|phasediff|epi|fieldmap)',
+        type_ = '(phase1|phase2|phasediff|epi|fieldmap)'
+        for file in self.get(subject=sub, type=type_,
                              extensions=['nii.gz', 'nii']):
-            metadata = self.get_metadata(file.filename)
+            metadata = self.get_metadata(file.filename, type=type_)
```

Comment:
Is it legal to pass in a target file that *could* exist, but doesn't? Because it looks like one of your tests (line 29) is doing that right now. I'm guessing it's not legal, and I should change this?